packages/transactions/src/prefix.ts

```diff
@@ -0,0 +1,36 @@
+import { Assignable } from '@near-js/types';
+
+const ACTIONABLE_MESSAGE_BASE = Math.pow(2, 30);
+// const NON_ACTIONABLE_MESSAGE_BASE = Math.pow(2, 31);
+
+/** The set of NEPs for which an [NEP-461](https://github.com/near/NEPs/pull/461) prefix is required on the message prior to hashing **/
+const NEP = {
+    MetaTransactions: 366,
+};
+
+/** Base class for NEP message prefixes **/
+abstract class NEPPrefix extends Assignable {
+    prefix: number;
+}
+
+/** Class for constructing prefixes on actionable (on-chain) messages **/
+abstract class ActionableMessagePrefix extends NEPPrefix {
+    /** Given the NEP number, set the prefix using 2^30 as the offset **/
+    protected constructor(prefix: number) {
+        super({ prefix: ACTIONABLE_MESSAGE_BASE + prefix });
+    }
+}
+
+/**
+ * Class for constructing prefixes on non-actionable (off-chain) messages
+ * @todo uncomment when off-chain messages are supported
```

Comment:
Is there an NEP, release version or date where we expect this to be true that we could reference here, or is this an indefinite 'some day' kind of thing? :)